tests/Http/HttpClientTest.php

```diff
@@ -666,11 +666,28 @@ public function testFilesCanBeAttached()
         $this->factory->assertSent(function (Request $request) {
             return $request->url() === 'http://foo.com/file' &&
                    Str::startsWith($request->header('Content-Type')[0], 'multipart') &&
-                   $request[0]['name'] === 'foo' &&
```

Comment:
Why was this test changed?

I understand the premise of the PR, but this test case is named "files can be attached", and the condition to check for an attached file is removed?

In the best case, no previous test cases should be changed, but again, I see from your proposed changes why you did it.

Either this test case should be renamed or removed.

Reply:
In this test, we check some of the other assertions related to attaching a file in a request, while in the new test, we focus solely on the functionality of the hasFile method itself. In my opinion, these tests complement each other and test different aspects of the framework. However, they could be merged into a single test that checks the entire process of attaching files.